bin/aws_cleaner.rb

```diff
@@ -26,84 +27,131 @@
+      AwsCleaner.new.delete_message(id, @config)
+    end
+  else
+    @logger.info("Instance #{@instance_id} does not exist in Chef, deleting message")
+    AwsCleaner.new.delete_message(id, @config)
+  end
+end
+
+def sensu(id)
+  chef_node = AwsCleaner::Chef.get_chef_node_name(@instance_id, @config)
+  if AwsCleaner::Sensu.in_sensu?(chef_node, @config)
+    if AwsCleaner::Sensu.remove_from_sensu(chef_node, @config)
+      @logger.info("Removed #{chef_node} from Sensu")
+    else
+      @logger.info("Instance #{@instance_id} does not exist in Sensu, deleting message")
+    end
+    AwsCleaner.new.delete_message(id, @config)
+  end
+end
+
+def closelog(message)
+  @logger.debug(message) unless message.nil?
+  @logger.close
+end
+
 # get options
 opts = Trollop.options do
   opt :config, 'Path to config file', type: :string, default: 'config.yml'
 end
 
 @config = config(opts[:config])
-
-# @sqs = Aws::SQS::Client.new(@config[:aws])
+@logger = logger(@config)
 @sqs_client = AwsCleaner::SQS.client(@config)
-
 @chef_client = AwsCleaner::Chef.client(@config)
 
 # main loop
 loop do
-  # get messages from SQS
-  messages = @sqs_client.receive_message(
-    queue_url: @config[:sqs][:queue],
-    max_number_of_messages: 10,
-    visibility_timeout: 3
-  ).messages
-
-  puts "Got #{messages.size} messages"
-
-  messages.each_with_index do |message, index|
-    puts "Looking at message number #{index}"
-    body = AwsCleaner.new.parse(message.body)
-    id = message.receipt_handle
-
-    unless body
-      AwsCleaner.new.delete_message(id, @config)
-      next
-    end
-
-    @instance_id = AwsCleaner.new.process_message(body)
-
-    if @instance_id
-      if @config[:webhooks]
-        @config[:webhooks].each do |hook, hook_config|
-          if AwsCleaner::Webhooks.fire_webhook(hook_config, @config, @instance_id)
-            puts "Successfully ran webhook #{hook}"
-          else
-            puts "Failed to run webhook #{hook}"
-          end
-        end
+  begin
+    # get messages from SQS
+    messages = @sqs_client.receive_message(
+      queue_url: @config[:sqs][:queue],
+      max_number_of_messages: 10,
+      visibility_timeout: 3
+    ).messages
+
+    @logger.info("Got #{messages.size} messages") unless messages.empty?
+
+    messages.each_with_index do |message, index|
+      @logger.info("Looking at message number #{index}")
+      body = AwsCleaner.new.parse(message.body)
+      id = message.receipt_handle
+
+      unless body
         AwsCleaner.new.delete_message(id, @config)
+        next
       end
 
-      chef_node = AwsCleaner::Chef.get_chef_node_name(@instance_id, @config)
+      @instance_id = AwsCleaner.new.process_message(body)
 
-      if chef_node
-        if AwsCleaner::Chef.remove_from_chef(chef_node, @chef_client, @config)
-          puts "Removed #{chef_node} from Chef"
-          AwsCleaner.new.delete_message(id, @config)
-        end
+      if @instance_id
+        webhook(id)
+        chef(id)
+        sensu(id)
```

Comment:
There's an issue here where the node will never get removed from sensu because the node is removed from chef first and we need the chef node object to get the node name. One way to solve this would be to set `@chef_node = AwsCleaner::Chef.get_chef_node_name(@instance_id, @config)` right before this section and then reference `@chef_node` in the methods. This would also avoid having to make multiple calls to chef to get the node name.

Reply:
I'll address all these when I got some free time. Thanks for reviewing.